homeassistant/components/honeywell/climate.py

```diff
@@ -311,6 +311,8 @@ async def _set_temperature(self, **kwargs) -> None:
 
         except AIOSomecomfort.SomeComfortError as err:
             _LOGGER.error("Invalid temperature %.1f: %s", temperature, err)
+        except TypeError as err:
```

Comment:
I suggest using `dict[key]` and just let it blow up instead. It's a developer problem if this happens and it should be fixed upstream.